packages/wmr/src/server.js

```diff
@@ -98,6 +98,11 @@ export default async function server({ cwd, root, overlayDir, middleware, http2,
 
 	app.ws = new WebSocketServer(app.server, '/_hmr');
 
+	// Custom middlewares should always come first, similar to plugins
```

Comment:
I'm wondering if we should have custom middleware come after the compression middleware. It modifies `res.write` & friends, but doing so after they've been called might break things.

Reply:
Good point, updated the PR